modules/rostests/winetests/ucrtbase/CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+
+# Hack until we support globally defining _DEBUG
+if(DBG)
+    #add_compile_definitions(_DEBUG)
+endif()
+
+# Generate threaddll.dll
+spec2def(threaddll.dll threaddll.spec)
+add_library(threaddll SHARED
+    threaddll.c
+    threaddll.def
+)
+set_entrypoint(threaddll DllMain 12)
+if(MSVC)
+    target_link_libraries(threaddll runtmchk)
+endif()
+add_importlibs(threaddll ucrtbase kernel32)
+
+# Hacks to make it work with MSVC IDE / MSBUILD
+if(MSVC_IDE)
+    if(DBG)
+        set(THREADDLL_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/Debug)
+    else()
+        set(THREADDLL_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/Release)
+    endif()
+    message(STATUS "threaddll.dll output directory: ${THREADDLL_OUTPUT_DIR}")
+
+    set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} /I${THREADDLL_OUTPUT_DIR}")
+endif()
+
+# Hack to replace the old CRT include directory with the UCRT include directory
+get_property(INCLUDE_DIRS DIRECTORY . PROPERTY INCLUDE_DIRECTORIES)
+list(REMOVE_ITEM INCLUDE_DIRS "${REACTOS_SOURCE_DIR}/sdk/include/crt")
```

Comment:
Instead of making this a global include, maybe we can add it as a public include to one of the `crt` libraries?

Reply:
Yes, but that is not simple, because basically everything uses these includes and you need to "fix" this everywhere. Also libraries are usually not linked to any CRT, so you'd need to set that manually, maybe somewhere in set_module_type or manually link stuff to an interface library like "crtheaders" vs "ucrtheaders". It's quite a bit of work.